### PR TITLE
Avoid unicode names when working with np.recarray in py2

### PR DESCRIPTION
**Pull request by jbrockmendel**
Fixes all but one of the test errors discussed in #3658

### CODE REVIEW

**Comment by jbrockmendel** [comment]
coveralls is reporting coverage decrease for python 3.6.  This is to be expected, as the relevant code added is conditioned on `is_py2`

File: statsmodels/compat/numpy.py [hunk 1]
```diff
@@ -87,6 +87,9 @@
 from .scipy import NumpyVersion
 import numpy as np
 
+import sys
+_is_py2 = sys.version_info.major == 2
```
Comment (bashtage) [inline statsmodels/compat/numpy.py]:
```from statsmodels.compat.python import PY3
if not PY3:
   ...
```

File: statsmodels/tsa/tsatools.py [hunk 2]
```diff
@@ -117,7 +122,19 @@ def safe_is_const(s):
         x = x.to_records(index=False, convert_datetime64=False)
         new_descr = x.dtype.descr
         extra_col = len(new_descr) - len(descr)
-        descr = new_descr[:extra_col] + descr if prepend else descr + new_descr[-extra_col:]
+        if prepend:
+            descr = new_descr[:extra_col] + descr
+        else:
+            descr = descr + new_descr[-extra_col:]
+
+        if _is_py2:
```
Comment (bashtage) [inline statsmodels/tsa/tsatools.py]: `not PY3` here too

**Comment by bashtage** [comment]
recarrays support should DIAF.

File: statsmodels/tsa/tsatools.py (hunk 2 above)
Comment (jbrockmendel) [inline statsmodels/tsa/tsatools.py]: Just pushed change.

**Comment by jbrockmendel** [comment]
> recarrays support should DIAF.

That's why this is a "fix the tests and move on with our lives" PR instead of a "carefully address the underlying issue" PR.

**Comment by josef-pkt** [comment]
I started to look a bit at the details of the related issues (including numpy and pandas)

I think we should try to fix recarrays one last time, and then deprecate the support of them to avoid "wasting" more time with numpy and pandas compat issues for structured dtypes and recarrays.
If possible we need to fix it for Debian #3841 and users who still are in python 2 land.

**Comment by jbrockmendel** [comment]
OK.  What do you have in mind for "try to fix"?

**Comment by josef-pkt** [comment]
You mentioned in the description that there is still one error that isn't fixed. I was planning to try out this PR and understand a bit better what is going on, and see whether we want bytes or string or a compat function in python 2 in the way or a similar way as you have in this PR.

Also, this PR needs a rebase on master which has as side effect that the tests will be run by pytest instead of nosetests.

(Because I had forgotten that I avoided the failing tests, I didn't realize before that this is high priority for 0.9, especially if numpy or pandas don't fix it for us.)

**Comment by josef-pkt** [comment]
(The cheapest solution would be to skip the failing unit tests on python 2, which would be the last resort if not everything is fixed.)

**Comment by josef-pkt** [comment]
Also, to see the results of this PR, we need to revert the two PRs that pin the pandas version on python 2  on travis and appveyor to one that "works".

**Comment by jbrockmendel** [comment]
OK.  I've stretched myself a bit too thin so will not be pro-active for a bit.  Please let me know if there's anything specific I can do to help.

**Comment by josef-pkt** [comment]
appveyor has upgraded to pandas 0.20.3 and the unicode errors are back
https://ci.appveyor.com/project/josef-pkt/statsmodels/build/1.0.2294/job/32pr6h5xh4jexanh

**Comment by jbrockmendel** [comment]
I've forgotten: is this one of the problems expected to resolve itself eventually?  Or one of the ones where we need to stop supporting X?

**Comment by josef-pkt** [comment]
I was planning to look into it this week and merge (I didn't get much done today)

My strategy:
- update pandas on travis and appveyor to see all errors again
- rebase this
- whatever tests keep failing will get a skiptest on python 2.7 (better unicode handling in statsmodels requires python 3)

It didn't look like it will go away by changes in pandas, because the problematic changes there were introduced to solve other problems. 
The problems for statsmodels will go away at some point in the future when we drop recarray and/or python 2.7 support.

**Pull request closed without merging**